### ISSUE TITLE
arrow-body-style: prefer body parens `(` 

### ISSUE THREAD

**Issue by peter-mouland** [eslint/eslint#10548]
**What rule do you want to change?**
arrow-body-style

**Does this change cause the rule to produce more or fewer warnings?**
more

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option

**Please provide some example code that this change will affect:**

```js
// simple example
(foo) =>
  my()
    .chaining()
    .function();

// real example
            encryptedData.map((encrypted) =>
                unzipAndDecryptAccount({
                    encryptedAccountData: encrypted.accountData,
                    encryptionKey: encryptionKeyDictionary[encrypted.accountId],
                    genericCryptoService,
                }),
            )
```
note the real example also has a comma dangle where it doesn't really belong

**What does the rule currently do for this code?**
This is perfectly valid and nothing changes. 

**What will the rule do after it's changed?**
My preference would be to say if it is multi-line, that brackets `(`) are used, and to not force braces `{`.

```js
// simple example
(foo) => (
  my()
    .chaining()
    .function()
);
//real example
            encryptedData.map((encrypted) => (
                unzipAndDecryptAccount({
                    encryptedAccountData: encrypted.accountData,
                    encryptionKey: encryptionKeyDictionary[encrypted.accountId],
                    genericCryptoService,
                })
            ))
```
plus, together with `implicit-arrow-linebreak` set to beside, the result before the change is:

```js
// simple example
(foo) => my()
  .chaining()
  .function();

// real example
            encryptedData.map((encrypted) => unzipAndDecryptAccount({
                encryptedAccountData: encrypted.accountData,
                encryptionKey: encryptionKeyDictionary[encrypted.accountId],
                genericCryptoService,
            }),
            )
```
note, the auto-fix has now left an orphaned bracket.

after the change applying `implicit-arrow-linebreak` set to beside both rules work together and don't conflict:

```js
// simple example
(foo) => (
  my()
    .chaining()
    .function()
);

// real example
            encryptedData.map((encrypted) => (
                unzipAndDecryptAccount({
                    encryptedAccountData: encrypted.accountData,
                    encryptionKey: encryptionKeyDictionary[encrypted.accountId],
                    genericCryptoService,
                })
            ))
```

**Comment by ilyavolodin** [comment]
Sorry, I'm not sure I understand what's wrong with this autofix example:

```js
            encryptedData.map((encrypted) => unzipAndDecryptAccount({
                encryptedAccountData: encrypted.accountData,
                encryptionKey: encryptionKeyDictionary[encrypted.accountId],
                genericCryptoService,
            }),
            )
```
It seems to be perfectly valid, unless I'm missing something.

**Comment by peter-mouland** [comment]
it is perfectly valid, and maybe it's just a preference thing which i need to get use to.  but i do prefer the version without the orphaned bracket and without the comma:

```js
            encryptedData.map((encrypted) => (
                unzipAndDecryptAccount({
                    encryptedAccountData: encrypted.accountData,
                    encryptionKey: encryptionKeyDictionary[encrypted.accountId],
                    genericCryptoService,
                })
            ))
```

**Comment by ljharb** [comment]
(`(` is a paren - `bracket` is both ambiguous and not universally used to describe parens/braces/square brackets/angle brackets)
